evaluation/plotting.py

```diff
@@ -0,0 +1,201 @@
+import os
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import matplotlib
+import numpy as np
+from matplotlib import pyplot as plt
+from matplotlib.figure import Figure
+from matplotlib.patches import RegularPolygon
+from matplotlib.path import Path as MatPath
+from matplotlib.projections import register_projection
+from matplotlib.projections.polar import PolarAxes
+from matplotlib.spines import Spine
+from matplotlib.transforms import Affine2D
+
+
+def save_figures(fig: Figure, output_path: Path, format: str = "pdf"):
+    """Save the current figure to the provided output path."""
+
+    if not isinstance(output_path, Path):
+        output_path = Path(output_path)
+
+    if not output_path.parent.exists():
+        os.mkdir(output_path.parent)
+
+    plt.savefig(output_path, bbox_inches="tight", format=format)
+    plt.close(fig)
```

Comment:
Is there a reason why you're specifically passing the figure here? It feels weird given that you're simply relying on the default when going through the rest of the operations (including `savefig` above).